Sources/Client/Client/Client.swift

```diff
@@ -18,16 +18,62 @@ public final class Client: Uploader {
     public typealias OnEvent = (Event) -> Void
     
     /// A client config (see `Config`).
-    public static var config = Config(apiKey: "")
-    /// A shared client.
-    public static let shared = Client()
+    @available(*, deprecated, message: """
+    Configuring the shared Client using the static `Client.config` variable has been deprecated. Please create an instance
+    of the `Client.Config` struct and call `Client.configure(_:)` to set up the shared instance of Client.
+    """)
+    public static var config: Config = .init(apiKey: "_deprecated") {
+        didSet {
+            guard _sharedClient == nil else {
+                assertionFailure("`Client.shared` instance was already used. It's not possible to change its configuration.")
```

Comment:
I think we should also log this, since it might happen on release builds and then when stuff goes awry there'll be no info...